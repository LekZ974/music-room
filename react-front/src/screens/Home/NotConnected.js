import React from 'react';
import PropTypes from 'prop-types';

import { StyleSheet } from 'react-native';
import I18n from '../../i18n';

import { Button, Container, Logo, View } from '../../components';

import { Theme } from '../../native-base-theme/default_theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    margin: Theme.spacing.small,
  },
});

const NotConnected = ({ navigation }) => (
  <Container style={styles.container}>
    <View>
      <Logo />
      <Button
        label={I18n.t('notConnected.loginButton')}
        style={styles.button}
        onPress={() => navigation.navigate('LoginScreen')}
        full
      />
      <Button
        label={I18n.t('notConnected.signUpButton')}
        style={styles.button}
        onPress={() => navigation.navigate('SignUpScreen')}
        full
      />
    </View>
  </Container>
);

NotConnected.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default NotConnected;
