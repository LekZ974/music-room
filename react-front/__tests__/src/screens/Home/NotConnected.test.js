import React from 'react';

import { shallow } from 'enzyme'
import NotConnected from '../../../../src/screens/Home/NotConnected';
import toJson from 'enzyme-to-json';

jest.mock('react-native-gesture-handler', () => {});
jest.mock('../../../../src/i18n', () => ({ t: jest.fn() }));

const navigation = { navigate: jest.fn() };

describe('>>>> NotConnected - REACT-REDUX (Shallow + passing the {store} directly', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NotConnected navigation={navigation} />)
  })
  it('renders without crashing', () => {
    expect(wrapper.length).toEqual(1)
  })
  it('Snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})