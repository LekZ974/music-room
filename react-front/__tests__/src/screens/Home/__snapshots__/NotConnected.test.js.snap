// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>> NotConnected - REACT-REDUX (Shallow + passing the {store} directly Snapshot 1`] = `
<Connect(Container)
  center={true}
  testID="homeNotConnected"
>
  <View>
    <Logo
      sm={false}
      source={null}
      testID="homeNotConnectedLogo"
    />
    <Button
      full={true}
      iconName={null}
      iconType={null}
      label={null}
      onPress={[Function]}
      style={
        Object {
          "margin": 16,
        }
      }
      testID="homeNotConnectedLoginButton"
    />
    <Button
      full={true}
      iconName={null}
      iconType={null}
      label={null}
      onPress={[Function]}
      style={
        Object {
          "margin": 16,
        }
      }
      testID="homeNotConnectedSignUpButton"
    />
  </View>
</Connect(Container)>
`;
